File: backend/src/utils/encrypt-password.js
import bcrypt from "bcryptjs";

const SALT_ROUNDS = 10;

class EncryptPassword {
    static async hashPassword(plainPassword) {
        return await bcrypt.hash(plainPassword, SALT_ROUNDS);
    }

    static async comparePassword(plainPassword, hashedPassword) {
        return await bcrypt.compare(plainPassword, hashedPassword);
    }
}

export default EncryptPassword;

File: backend2/internal/util/encrypt_password.go
package util

import "golang.org/x/crypto/bcrypt"

const saltRounds = 10

func HashPassword(plain string) (string, error) {
	hashed, err := bcrypt.GenerateFromPassword([]byte(plain), saltRounds)
	if err != nil {
		return "", err
	}
	return string(hashed), nil
}

func ComparePassword(plain, hashed string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(hashed), []byte(plain))
	return err == nil
}
